File: pclassic/2016f/stubs/Tricky.java
import java.util.*;
import java.io.*;
public class Tricky {
    public static void main(String[] args) throws FileNotFoundException {
        // Before submitting, make sure the main method hasn't been changed!
        Scanner sc = new Scanner(new FileReader("TrickyIN.txt"));
        while(sc.hasNext()) {
            int length = sc.nextInt();
            Iterable<Integer> output = findTriples(length);
            String s = "[";
            boolean isFirstElement = true;
            for(Integer i : output) {
                if(isFirstElement) {
                    s += i;
                    isFirstElement = false;
                }
                else {
                    s += ", "+ i;
                }
            }
            s += "]";
            System.out.println(s);   
        }
        sc.close();
    }
    
    public static Iterable<Integer> findTriples(int hypotenuse) {
        /** Implement this method.
         
            This method returns a list containing all possible configurations of the other two
            sides that are of positive integer length. Output each
            configuration as a separate element in a list in the format a b
            where a is in ascending order and b is in descending order in
            respect to the other configurations.
             
            hypotenuse - the length of the aqueduct you want to construct.
         
        **/ 
        return null;
    }
}

File: pclassic/2016f/stubs/Tricky.py
def is_perfect_square(n):
    return int(n ** 0.5) ** 2 == n

def pythagorean_triples(c):
    """

    :param c: the length of the aqueduct you want to construct.
    :return: a list containing all possible configurations of the other two
             sides that are of positive integer length. Output each
             configuration as a separate element in a list in the format a b
             where a is in ascending order and b is in descending order in
             respect to the other configurations.
    """

    # TODO: implement

    ret = []

    for a in range(1, c):
        sq = c ** 2 - a ** 2
        
        if is_perfect_square(sq):
            b = int(sq ** 0.5)
            ret.append(a)
            ret.append(b)
        
    return ret

def parse_file_and_call_function():
    with open("TrickyIN.txt", "r") as f:
        for line in f:
            print list(pythagorean_triples(int(line)))

if __name__ == "__main__":
    parse_file_and_call_function()
